packages/hub/services/search-index.ts

```diff
@@ -240,6 +243,17 @@ class IndexerRun implements IndexerHandle {
     } while (rows.length > 0);
   }
 
+  async read(card: RawCard): Promise<RawCard | void> {
+    try {
+      let rawCard = await this.realmManager.read(card);
+      return rawCard;
+    } catch (err: any) {
+      let compiler = await this.builder.compileCardFromRaw(card);
```

Comment:
This isn't necessary. When we have an error during read, there are no external dependencies that would cause us to need to try to index the card again. Only changes to the card itself should cause a retry. So you can pass an empty Set to `saveErrorState` below.